Habitat Features Branch: Support Sampling Information

NickPhura opened the pull request:
Add basic sampling data to the existing find/get survey habitat features endpoints.

Add sampling form/controls to the create/edit survey habitat feature page.

NickPhura committed:
DB changes to add sample period to habitat features.

NickPhura committed:
API changes to add sample period to habitat features.

NickPhura committed:
APP changes to add sample period to habitat features.

NickPhura committed:
Include site and technique information in get habitat feature response

NickPhura committed:
Update create/edit habitat feature forms to support sampling

NickPhura committed:
ignore-skip

MacDeluca reviewed (commented):
Code looks great!

Some small things I found when testing the app:
1. Looks like the font colour of the inputs is slightly different than our other inputs
![image](https://github.com/user-attachments/assets/182b14a0-b99c-4ca5-92e0-ca34085be6be)

2. The table doesn't appear to be updating when I edit an existing habitat feature (I added sample site metadata)

3. If I select Sample Site 2 and try to select a period before a technique it clears after I select it. I think this is cause it doesnt have a method technique? http://localhost:7100/admin/projects/1/surveys/1/habitat-features/1/edit

3. Note: Unrelated to your PR. If you manually clear the "Observed Date" field it sets to "Invalid Date"

NickPhura committed:
Fix sampling information controls UI.
Fix sampling information controls options when selecting/unselecting.
Add yup schema validation for date + lat/lng vs sample period.
Fix Invalid Date issue when clearing observed_date field.
Bonus: Removed edit/remove map controls from animal capture/mortality (as they are unused and didnt do anything).

NickPhura commented:
> Code looks great!
> 
> Some small things I found when testing the app:
> 
>     1. Looks like the font colour of the inputs is slightly different than our other inputs
> 
>     2. The table doesn't appear to be updating when I edit an existing habitat feature (I added sample site metadata)
> 
>     3. If I select Sample Site 2 and try to select a period before a technique it clears after I select it. I think this is cause it doesnt have a method technique? http://localhost:7100/admin/projects/1/surveys/1/habitat-features/1/edit
> 
>     4. Note: Unrelated to your PR. If you manually clear the "Observed Date" field it sets to "Invalid Date"

I believe I've addressed all of these issues in my most recent commit. 🦎

NickPhura committed:
Add required flag to count field

MacDeluca reviewed (approved):
🤔
